zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -994,15 +993,12 @@ Either<ProblemDetail, T> buildSearchQuery(
             queryBuilderSupplier.get().page(page.get()).filter(filter).sort(sorting.get()).build());
   }
 
-  private static List<String> applySortOrder(
-      final String order, final SortOption.AbstractBuilder<?> builder) {
-    final List<String> validationErrors = new ArrayList<>();
-    switch (order.toLowerCase()) {
-      case "asc" -> builder.asc();
+  private static void applySortOrder(
+      final SortOrderEnum order, final SortOption.AbstractBuilder<?> builder) {
+    switch (order.getValue()) {
       case "desc" -> builder.desc();
```

Comment:
🔧 I guess you can do a proper switch on the enum here, right?
```suggestion
    switch (order) {
      case DESC -> builder.desc();
```